crates/c-api/include/wasi.h

```diff
@@ -39,6 +39,45 @@ extern "C" {
  */
 WASI_DECLARE_OWN(config)
 
+/**
+ * \typedef wasi_read_pipe_t
+ * \brief Convenience alias for #wasi_read_pipe_t
+ *
+ * \struct wasi_read_pipe_t
+ * \brief Reader end of a bounded FIFO buffer
+ * 
+ * An instance using fd_read on an empty pipe with succeeded but with 0 bytes read.
+ * The effect depends on the implementation. For example, Rust's
+ * `std::io::Read::read` returns Ok(0), while `read_exact` returns Err
+ * 
+ * It is safe to move a #wasi_read_pipe_t to any thread at any time.
+ * However concurrent interactions may lock.
```

Comment:
I think it's ok to remove this "However ..." clause because pipes leading to deadlock aren't necessarily unique to WASI.